jupyterhub/utils.py

```diff
@@ -247,6 +247,13 @@ def admin_only(self):
     if user is None or not user.admin:
         raise web.HTTPError(403)
 
+@auth_decorator
+def metrics_authentication(self):
+    """Decorator for restricting access to metrics"""
+    user = self.current_user
+    if user is None and not self.authenticate_prometheus:
```

Comment:
This should be `and self.authenticate_prometheus`. The idea here is that `if self.authenticate_prometheus`, we should only allow authenticated users.